File: server/src/main/java/fr/bookhub/entity/Review.java
package fr.bookhub.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @NotNull
    private Book book;

    @ManyToOne
    @NotNull
    private User user;
    private Integer rating;

    @Lob
    private String comment;

    private Boolean isHidden = false;

    @ManyToOne
    @JoinColumn(name = "hidden_by")
    private User hiddenBy;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "updated_by")
    private User updatedBy;
}

File: web/src/app/core/type/review.ts
export interface Review {
  reviewId?: number;
  bookId: number;
  user?: UserBasicResponse;
  rating: number;
  comment: string;
  isHidden?: boolean;
  hiddenBy?: UserBasicResponse;
  createdAt?: string;
  updatedAt?: string;
  updatedBy?: UserBasicResponse;
}

export interface UserBasicResponse {
  id: number;
  username: string;
  firstName?: string;
  lastName?: string;
}
